### ISSUE TITLE
Callback support

### ISSUE THREAD

**Issue by amirhouieh** [smith-kyle/electron-process#1]
Hey, this module is the one which I needed! TNX! but I've got a problem with catching asynchronous function's response, here is what I am trying to do:

in foreground:

```
const foreground = require('electron-process').foreground;
const modulesController = foreground.getModule(require('./mybackgroundmodule'));

modulesController.readProfiles()
    .then(function (profiles) {
        //do some stuff with profiles
    });
```

in background

```
const profiles = [];
const Controller = {

    readProfiles(callback) {

        fse.walk("./somedirectory")
            .on('data', function appendModule(item) {
                profiles.push(fse.readJsonSync(item.path, {throw: false}));
            })
            .on('end', function () {
                //here I am done with reading all data so 
                //i would expect something like:
                callback(profiles);
            });
    }
};
module.exports = Controller;
```

But this just does not work! I am quite new with js 6 and promises, so any hint would be appreciated! 
And actually i would export my background module on my own way, but again this does not working neither!

in forground:

```
const foreground = require('electron-process').foreground;
const profileController = foreground.getModule(require('./mybackgroundmodule'));

    profileController.readProfiles(function () {

        profileController.each(function(m,profile) {
                //do stuff with each profile
        })

    })
```

in background:

```
module.exports = Controller;

...

function Controller() {

    const self = this;
    self.profiles = [];

   //method #1
    self.readProfiles = function (callback) {

        fse.walk(modulesDir)
            .on('data', function(item) {
                var m = new ProfileSchema(item.path);
                self.profiles.push(m);
            })
            .on('end', function () {
                if(callback)
                    callback(self.profiles);
            });
    };

   //method #2
    self.each = function (onEach) {
        for(var x in self.profiles)
            onEach(x,self.profiles[x]);
    }

    ...
    //method #n

    return this;
}

...

```

**Comment by smith-kyle** [comment]
Did an error get thrown? Or did the promise returned from `readProfiles()` not resolve?

Make sure to register the `Controller.js` module on the background with `electronProcess.background.registerModule`

**Comment by amirhouieh** [comment]
I was getting nothing! my problem was more about how to pass callback function to the background module. Any way I got over it by using walkSync instead, so the Controller now looks like this:

```
const Controller = {
    readProfiles() {
        let profilePaths = walkSync(profiles_dir, {globs: ['*/profile.json'], directories: false });
        return processProfiles(profilePaths);
    }
};

```

But still I am looking forward to know how to manage callbacks! tnx

**Comment by smith-kyle** [comment]
So callbacks actually do not work between the foreground and the background. At least right now. It's definitely something I'd like to address in the future.

**Comment by amirhouieh** [comment]
Looking forward to it! and indeed willing to contribute as soon as I get time!

**Comment by smith-kyle** [comment]
Added in #3. @amirhouieh Hopefully that works for you.

**Comment by smith-kyle** [comment]
Published as `0.2.0`
